.github/workflows/tests.yaml

```diff
@@ -302,16 +302,16 @@ jobs:
         if-no-files-found: ignore
 
     - name: collect test coverage
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
       with:
-        name: coverage
+        name: coverage-integration-webhook-enabled-${{ matrix.webhook-enabled }}
         path: coverage.integration.out
 
     - name: collect test report
       if: ${{ always() }}
-      uses: actions/upload-artifact@v3
+      uses: actions/upload-artifact@v4
```

Comment:
What about pinning to a specific commit? Are we going to address it everywhere in a separate PR? Btw. unfortunately I don't see such check in a linter to enforce pinning to a commit

Reply:
There's a contribution to the linter that should allow verifying it. Looks stale though: https://github.com/rhysd/actionlint/pull/436

Reply:
Good idea. I've substituted the tags with digests in 741a8b363ba2f15dd1c986d5551588cf9cdc7558.

That was done using:

```
GH_ADMIN_TOKEN=${GITHUB_TOKEN} pin-github-action --ignore-shas --comment " {ref}" .github/workflows/tests.yaml .github/workflows/__build-workflow.yaml .github/workflows/__release-workflow.yaml .github/workflows/release-bot.yaml .github/workflows/build.yaml .github/workflows/release.yaml .github/workflows/backport.yaml .github/workflows/nightly.yaml
```